packages/flutter_tts/tizen/src/flutter_tts_tizen_plugin.cc

```diff
@@ -137,74 +77,48 @@ class FlutterTtsTizenPlugin : public flutter::Plugin {
     const auto method_name = method_call.method_name();
     const auto &arguments = *method_call.arguments();
     LOG_INFO("method : %s", method_name.c_str());
-    int ret = TTS_ERROR_NONE;
 
     if (method_name.compare("awaitSpeakCompletion") == 0) {
       if (std::holds_alternative<bool>(arguments)) {
-        await_speak_completion_ = std::get<bool>(arguments);
+        bool await_speak_completion = std::get<bool>(arguments);
+        tts_->SetAwaitSpeakCompletion(await_speak_completion);
         result->Success(flutter::EncodableValue(1));
         return;
       }
       result->Success(flutter::EncodableValue(0));
     } else if (method_name.compare("speak") == 0) {
-      if (speaking_) {
+      if (tts_->GetState() == TTS_STATE_PLAYING) {
         LOG_ERROR("[TTS] : You cannot speak again while speaking.");
         result->Success(flutter::EncodableValue(0));
         return;
       }
 
       if (std::holds_alternative<std::string>(arguments)) {
         std::string text = std::get<std::string>(arguments);
-        int utt_id;
-        ret = tts_add_text(tts_, text.c_str(), language_.c_str(), voice_type_,
-                           speed_, &utt_id);
-        if (ret != TTS_ERROR_NONE) {
-          LOG_ERROR("[TTS] tts_add_text failed: %s", get_error_message(ret));
+        if (tts_->AddText(text)) {
+          tts_->Speak(std::move(result));
+        } else {
           result->Success(flutter::EncodableValue(0));
```

Comment:
Why is Success() called instead of Error() when AddText() is failed?

Reply:
I followed the original creator's intent and imitated the Android implementation.

Please, see for detail: 
- https://github.com/flutter-tizen/plugins/pull/152 
- https://github.com/flutter-tizen/plugins/issues/149

Reply:
I understood it.